### ISSUE TITLE
[Bug] Generate rules should specify apiVersion in its definition

### ISSUE THREAD

**Issue by shubham4443** [kyverno/policies#294]
### Kyverno Version

1.5.x

### Kubernetes Version

1.23.x

### Kubernetes Platform

Other (specify in description)

### Description

Generate policies will be covered by the test command soon. See PR kyverno/kyverno#3456. The CLI requires that user mentions apiVersion in generate policy's rule definition otherwise the test command would print out wrong result. Additionally, it is best practice to specify apiVersion along with kind.

### Steps to reproduce

1. NA

### Expected behavior

NA

### Troubleshooting

- [X] I have read and followed the documentation AND the [troubleshooting guide](https://kyverno.io/docs/troubleshooting/).
- [X] I have searched other issues in this repository and mine is not recorded.

**Comment by JimBugwadia** [comment]
Seems like `apiVersion` is also required for any CRD. Otherwise we see a cryptic error message:

```
    message: 'Internal error occurred: /, Kind=PipelineRun is unstructured and is
      not suitable for converting to "tekton.dev/v1beta1"'
```

We need to update the docs for this, and also improve the error message to indicated that the `apiVersion` is required.

**Comment by chipzoller** [comment]
> Seems like `apiVersion` is also required for any CRD. Otherwise we see a cryptic error message:

Is this for any type of Kyverno rule? I can create a docs issue but need to understand the change.

**Comment by JimBugwadia** [comment]
No, for generate rules only.

**Comment by chipzoller** [comment]
https://github.com/kyverno/website/issues/502

**Comment by chipzoller** [comment]
Closed in #314
